### ISSUE TITLE
Android build script doesn't work on macOS

### ISSUE THREAD

**Issue by thomcc** [mozilla/mentat#749]
The version of bash it requires is too new. It's possible to request that users install a newer bash but we should probably just make it not require that.

**Comment by ncalexan** [comment]
> The version of bash it requires is too new. It's possible to request that users install a newer bash but we should probably just make it not require that.

Yes, I agree.  I didn't think it actually had a functioning shebang, but maybe that's some other script in there.  I've always ran it as `bash ...` anyway, since I run Fish instead.

**Comment by thomcc** [comment]
Sadly, running as `bash ...` still requires a sufficently new bash, which is the bug I'm reporting. I could make it work by updating the version of bash I have installed (and I probably will eventually), but IMO the amount we gain by requiring that is minimal enough not to be worth it.

(FYI since I also use fish, you can still run executable shell scripts with shebangs as ./blah, even if they aren't written in fish).

**Comment by grigoryk** [comment]
Ah, this is annoying. I used bash 4.0's associative arrays, which I think is what's causing the problem. 4.0 was announced nearly 10 years ago - but apparently since it's licensed under GPLv3, OSX won't include it and instead uses an older GPLv2 version.

We should change the script :(
